<a href="https://colab.research.google.com/github/pattmikyu/wine-quality-classification/blob/main/scripts_aux_tech_challenge.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# importando as bibliotecas necessárias para análises
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt


In [ ]:
df = pd.read_csv('WineQT.csv')

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1143 entries, 0 to 1142
Data columns (total 13 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         1143 non-null   float64
 1   volatile acidity      1143 non-null   float64
 2   citric acid           1143 non-null   float64
 3   residual sugar        1143 non-null   float64
 4   chlorides             1143 non-null   float64
 5   free sulfur dioxide   1143 non-null   float64
 6   total sulfur dioxide  1143 non-null   float64
 7   density               1143 non-null   float64
 8   pH                    1143 non-null   float64
 9   sulphates             1143 non-null   float64
 10  alcohol               1143 non-null   float64
 11  quality               1143 non-null   int64  
 12  Id                    1143 non-null   int64  
dtypes: float64(11), int64(2)
memory usage: 116.2 KB


A variável de qualidade deverá ser transformada em uma classificação
binária

In [ ]:
df.head(10)

,fixed acidity,volatile acidity,citric acid,residual sugar,chlorides,free sulfur dioxide,total sulfur dioxide,density,pH,sulphates,alcohol,quality,Id
0,7.4,0.70,0.00,1.9,0.076,11.0,34.0,0.9978,3.51,0.56,9.4,5,0
1,7.8,0.88,0.00,2.6,0.098,25.0,67.0,0.9968,3.20,0.68,9.8,5,1
2,7.8,0.76,0.04,2.3,0.092,15.0,54.0,0.9970,3.26,0.65,9.8,5,2
3,11.2,0.28,0.56,1.9,0.075,17.0,60.0,0.9980,3.16,0.58,9.8,6,3
4,7.4,0.70,0.00,1.9,0.076,11.0,34.0,0.9978,3.51,0.56,9.4,5,4
5,7.4,0.66,0.00,1.8,0.075,13.0,40.0,0.9978,3.51,0.56,9.4,5,5
6,7.9,0.60,0.06,1.6,0.069,15.0,59.0,0.9964,3.30,0.46,9.4,5,6
7,7.3,0.65,0.00,1.2,0.065,15.0,21.0,0.9946,3.39,0.47,10.0,7,7
8,7.8,0.58,0.02,2.0,0.073,9.0,18.0,0.9968,3.36,0.57,9.5,7,8
9,6.7,0.58,0.08,1.8,0.097,15.0,65.0,0.9959,3.28,0.54,9.2,5,10


In [ ]:
#quais notas possuimos na coluna quality (primeiro vamos transformar em int)
df['quality'] = df['quality'].astype(int)

unique_n= df['quality'].unique()
print(unique_n)

[5 6 7 4 8 3]


In [ ]:
df.describe()

,fixed acidity,volatile acidity,citric acid,residual sugar,chlorides,free sulfur dioxide,total sulfur dioxide,density,pH,sulphates,alcohol,quality,Id
count,1143.000000,1143.000000,1143.000000,1143.000000,1143.000000,1143.000000,1143.000000,1143.000000,1143.000000,1143.000000,1143.000000,1143.000000,1143.000000
mean,8.311111,0.531339,0.268364,2.532152,0.086933,15.615486,45.914698,0.996730,3.311015,0.657708,10.442111,5.657043,804.969379
std,1.747595,0.179633,0.196686,1.355917,0.047267,10.250486,32.782130,0.001925,0.156664,0.170399,1.082196,0.805824,463.997116
min,4.600000,0.120000,0.000000,0.900000,0.012000,1.000000,6.000000,0.990070,2.740000,0.330000,8.400000,3.000000,0.000000
25%,7.100000,0.392500,0.090000,1.900000,0.070000,7.000000,21.000000,0.995570,3.205000,0.550000,9.500000,5.000000,411.000000
50%,7.900000,0.520000,0.250000,2.200000,0.079000,13.000000,37.000000,0.996680,3.310000,0.620000,10.200000,6.000000,794.000000
75%,9.100000,0.640000,0.420000,2.600000,0.090000,21.000000,61.000000,0.997845,3.400000,0.730000,11.100000,6.000000,1209.500000
max,15.900000,1.580000,1.000000,15.500000,0.611000,68.000000,289.000000,1.003690,4.010000,2.000000,14.900000,8.000000,1597.000000


In [ ]:
# média (soma de todas as notas dividida pela quantidade)
media = df['quality'].mean()
print(f"Média: {media:.2f}")

# mediana (o valor que divide os dados exatamente ao meio)
mediana = df['quality'].median()
print(f"Mediana: {mediana}")

# moda (a nota que mais se repete)
# O [0] no final é necessário porque pode existir mais de uma moda, então pegamos a primeira
moda = df['quality'].mode()[0]
print(f"Moda: {moda}")

Média: 5.66
Mediana: 6.0
Moda: 5


com essa conclusão, podemos manter a nota para vinhos de alta qualidade como >= 7. Pois, se a moda é 5.0 e a média é 5.66, significa que o vinho 'padrão' da base é nota 5.
Como a mediana é 6.0, pelo menos 5% dos vinhos são nota 6 para baixo.
Como as notas estao indo de 1 em 1, para diferenciar dos vinhos mediocres, a próxima nota seria 7.

In [ ]:
df.groupby(['volatile acidity']).describe()

fixed acidity                                              \
                         count    mean       std  min    25%   50%     75%   
volatile acidity                                                             
0.12                       2.0  8.9000  0.000000  8.9  8.900  8.90   8.900   
0.16                       1.0  6.7000       NaN  6.7  6.700  6.70   6.700   
0.18                       8.0  9.2625  1.761442  7.7  7.900  8.25  11.200   
0.19                       1.0  7.9000       NaN  7.9  7.900  7.90   7.900   
0.20                       2.0  8.0500  0.212132  7.9  7.975  8.05   8.125   
...                        ...     ...       ...  ...    ...   ...     ...   
1.07                       1.0  7.3000       NaN  7.3  7.300  7.30   7.300   
1.09                       1.0  6.9000       NaN  6.9  6.900  6.90   6.900   
1.18                       1.0  8.0000       NaN  8.0  8.000  8.00   8.000   
1.33                       2.0  8.1500  0.070711  8.1  8.125  8.15   8.175   
1.58                       1.0  7.6000       NaN  7.6  7.600  7.60   7.600   

                       citric acid        ... quality         Id            \
                   max       count  mean  ...     75%  max count      mean   
volatile acidity                          ...                                
0.12               8.9         2.0  0.45  ...    7.00  7.0   2.0   949.500   
0.16               6.7         1.0  0.64  ...    6.00  6.0   1.0  1566.000   
0.18              11.5         8.0  0.44  ...    6.00  7.0   8.0   898.625   
0.19               7.9         1.0  0.42  ...    6.00  6.0   1.0  1087.000   
0.20               8.2         2.0  0.39  ...    6.75  7.0   2.0  1302.000   
...                ...         ...   ...  ...     ...  ...   ...       ...   
1.07               7.3         1.0  0.09  ...    5.00  5.0   1.0   120.000   
1.09               6.9         1.0  0.06  ...    4.00  4.0   1.0   199.000   
1.18               8.0         1.0  0.21  ...    5.00  5.0   1.0  1312.000   
1.33               8.2         2.0  0.00  ...    5.00  5.0   2.0   126.500   
1.58               7.6         1.0  0.00  ...    3.00  3.0   1.0  1299.000   

                                                                        
                         std     min      25%     50%      75%     max  
volatile acidity                                                        
0.12                0.707107   949.0   949.25   949.5   949.75   950.0  
0.16                     NaN  1566.0  1566.00  1566.0  1566.00  1566.0  
0.18              532.401282   269.0   295.00  1087.0  1279.75  1509.0  
0.19                     NaN  1087.0  1087.00  1087.0  1087.00  1087.0  
0.20              222.031529  1145.0  1223.50  1302.0  1380.50  1459.0  
...                      ...     ...      ...     ...      ...     ...  
1.07                     NaN   120.0   120.00   120.0   120.00   120.0  
1.09                     NaN   199.0   199.00   199.0   199.00   199.0  
1.18                     NaN  1312.0  1312.00  1312.0  1312.00  1312.0  
1.33                0.707107   126.0   126.25   126.5   126.75   127.0  
1.58                     NaN  1299.0  1299.00  1299.0  1299.00  1299.0  

[135 rows x 96 columns]

In [ ]:
# Conta a quantidade de valores nulos em cada coluna
valores_nulos = df.isnull().sum()
print(valores_nulos)

fixed acidity           0
volatile acidity        0
citric acid             0
residual sugar          0
chlorides               0
free sulfur dioxide     0
total sulfur dioxide    0
density                 0
pH                      0
sulphates               0
alcohol                 0
quality                 0
Id                      0
dtype: int64
